### PR TITLE
Loosen zstd 1.4.x pin

### PR DESCRIPTION
**Pull request by xhochy**
Checklist
* [x] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by jjhelmus** [comment]
Thanks @xhochy, looks good to me.

**Comment by beckermr** [comment]
Can we post a diff for this one?

**Comment by xhochy** [comment]
Diff: https://gist.github.com/d662b355113b3af051ee224113a3fed1

File: recipe/gen_patch_json.py [hunk 1]
```diff
@@ -568,17 +568,16 @@ def _gen_new_index(repodata, subdir):
...
 
-        if 'snappy >=1.1.7,<1.1.8.0a0' in deps:
-            i = record['depends'].index('snappy >=1.1.7,<1.1.8.0a0')
-            record['depends'][i] = 'snappy >=1.1.7,<2.0.0.0a0'
+        _replace_pin('zstd >=1.4.0,<1.4.1.0a0', 'zstd >=1.4.0,<1.5.0.0a0', deps, record)
+        _replace_pin('zstd >=1.4.1,<1.4.2.0a0', 'zstd >=1.4.1,<1.5.0.0a0', deps, record)
+        _replace_pin('zstd >=1.4.2,<1.4.3.0a0', 'zstd >=1.4.2,<1.5.0.0a0', deps, record)
+        _replace_pin('zstd >=1.4.3,<1.4.4.0a0', 'zstd >=1.4.3,<1.5.0.0a0', deps, record)
+        _replace_pin('zstd >=1.4.4,<1.4.5.0a0', 'zstd >=1.4.4,<1.5.0.0a0', deps, record)
+        _replace_pin('zstd >=1.4.5,<1.4.6.0a0', 'zstd >=1.4.5,<1.5.0.0a0', deps, record)
```
Comment (isuruf) [inline recipe/gen_patch_json.py:576]: Can we use loosen_pin helper function

Comment (xhochy) [reply recipe/gen_patch_json.py:576]: @isuruf To what are you referring? I don't see such a function in the repo.

**Comment by beckermr** [comment]
This diff is a monster. Is there a specific package you are interested in? Should we only be patching that one?

**Comment by xhochy** [comment]
We should either patch all or do a rebuild. Otherwise we will get conflicts. I'm using several of the listed packages in my environments, only touching a selection is a not viable option.

**Pull request merged**